Issue guidelines review (#7)
mdn/temp-processes: issues.md

---
title: Guidelines to open, triage, and work on issues
slug: MDN/Community/Issues
page-type: mdn-community-guide
tags:
  - meta
  - community-guidelines
  - governance
---

{{MDNSidebar}}

As a contributor, you can [report](#Guidelines_for_reporting_an_issue) and [work](#guidelines_for_working_on_an_issue) on issues.

After you report an issue, the issue gets triaged. Issue [triaging](#guidelines_for_triaging_issues) is typically done by people assigned the role of a maintainer or an owner.

## General guidelines for participation

While reporting an issue or participating in a conversation in an issue, always ensure that your inputs are contributing to the overall progress of the project. Consider whether the issues you open and your comments in an issue are constructive and on topic and are not just adding noise.

Do the following:

- Before filing an issue, consider if you need to start a [discussion](https://github.com/mdn/mdn-community/discussions) in the MDN project on GitHub. Use discussions to gain different viewpoints and to converge on an agreed upon course of action. This helps to keep issues focused and productive.

> - Before filing an issue, read our [contribution guide](https://github.com/mdn/content/blob/main/CONTRIBUTING.md) to try to fix the problem yourself.

Reviewer: ```suggestion
- Before filing an issue, read our [contribution guide](https://github.com/mdn/content/blob/main/CONTRIBUTING.md) to try to fix the problem yourself.
```
Not sure about the wording here. It almost sounds like we are saying you could open a pull request instead of an issue. We want to encourage people to file an issue so it is tracked, but we also want to encourage them to contribute if they can. We could rephrase it a little to make that clear. What do you think?

Author: Definitely and good catch, thanks! We are saying that later on - to open an issue and see if you can try to fix it. So it should say the same here.

- If you have a question, you can ask using mechanisms like [chat rooms](https://chat.mozilla.org/#/room/#mdn:mozilla.org) or [forums](https://discourse.mozilla.org/c/mdn/236) without filing an issue.

Avoid doing the following:

- Complicating issues by trying to discuss multiple topics or by making off-topic comments.
- Opening lots of issues asking vague questions.

> - Asking questions without trying to solve the problem yourself first.

Reviewer: Here again, we can reword this a little. If you know what the problem is, opening an issue is a good first step. We, however, welcome contributions and so if you can, opening a pull request referencing the issue, would be appreciated.

Author: There is a small hurdle here and we might want to discuss this offline.
We've mentioned that before assigning an issue to themselves, contributors must wait for the issue to be triaged. I assume this includes issues contributors open and want to fix themselves. They'll need to wait for the triage step before opening the pull request.
See the updated text in https://github.com/mdn/temp-processes/pull/7/files lines 117-119:
```
In general, begin by opening the issue.
Add a comment about your intent to work on the issue.
After the issue is triaged, assign the issue to yourself.
Using the guidelines on working on an issue, try to fix the problem...
```
I think the triage step is important because we would not like contributors to open pull requests for issues that we don't want to be fixed or not in the way proposed.
(This actually reminds me to add a note in triaging to check for the validity of an issue.)

## Guidelines for reporting an issue

> [Issues](https://docs.github.com/en/github/managing-your-work-on-github/about-issues) are used to track bugs. An issue must be a single actionable task or a collection of multiple, related actionable tasks and must have a clear actionable outcome.

Reviewer 2: ```suggestion
[Issues](https://docs.github.com/en/github/managing-your-work-on-github/about-issues) are used to track bugs. An issue must be a single actionable task or a collection of related actionable tasks and must have a clear outcome.
```

### Before filing an issue

> If you've found a bug with either the content on MDN or the platform, search the current open issues in the [relevant repository](/en-US/docs/MDN/Community/Contributing/Our_repositories) to ensure that someone has not already reported the issue.

Reviewer 2: I would suggest it's more like "verify" first:

```suggestion
If you think you've found a bug with the content on MDN or a display problem, search the current open issues in the [relevant repository](/en-US/docs/MDN/Community/Contributing/Our_repositories) and make sure nobody else has reported the issue.
```

Reviewer 2: Also in general I think we can refer to "the platform" as "display issues" to make it easier to understand

Reviewer: Another item is that we should use MDN Web Docs consistently when referring to the project. It is easy to forget as one tends to refer to it as MDN 😸

Author: @bsmth good idea to clarify the meaning of 'platform'. I'm inclined to use '[look and feel](https://en.wikipedia.org/wiki/Look_and_feel)' to cover both display and functionality.

### Reporting an issue

> - Depending on the problem you've discovered, report it by filing a [documentation issue](https://github.com/mdn/content/issues), a [localization issue](https://github.com/mdn/translated-content/issues), or a [platform issue](https://github.com/mdn/yari/issues).

Reviewer: ```suggestion
- Depending on the problem you've discovered, report it by filing a [documentation issue](https://github.com/mdn/content/issues), a [localization issue](https://github.com/mdn/translated-content/issues), or a [frontend code issue](https://github.com/mdn/yari/issues).
```

Maybe?

Author: Do you think the term "look and feel" would be easier to understand? Brian also brought up the problem with the word "platform". See https://github.com/mdn/temp-processes/pull/7#discussion_r1062874333.

Author: Do you think the term "look and feel" will be easier to understand?
Brian also brought up the problem with the word "platform": https://github.com/mdn/temp-processes/pull/7#discussion_r1062874333

- To open an issue, use the appropriate template available in the repository. For example, to report a content bug, use the the [Content issue](<https://github.com/mdn/content/issues/new?assignees=&labels=needs+triage&template=content-bug.yml>) template in the `mdn/content` repository.

> - Provide sufficient information while reporting the issue:

Reviewer: Should we add this to the point of interaction? In other words, should we add this to the issue template directly? It might just be me :), but I love having this information when doing the task. It prevents me from only later learning that there are guidelines I should have followed.

Reviewer: We might not be as verbose as here, but we can link to this within the issue template. Something like, "For additional guidance on filing issues, please refer to [our documentation](link)."

Author: This is a great idea! And I agree, this information should be available right where/when doing the task.

I've added a To Do in the [issue description](https://github.com/mdn/temp-processes/pull/7#issue-1502294140) as a reminder:
`- [ ] Add link to content issue template to the guidelines here for "Reporting an issue"`

  - **Issue title** must convey succinctly the _required action_.

  - **Issue description** must clearly describe the bug and the action required to resolve the issue. It must also list the task or sub-tasks to be completed to resolve the issue. Some other guidelines include:
    - Use the description field to indicate the status of the task or sub-tasks by using checklists.

>     - Updates to the status of a task should be conveyed by editing the task list in the issue description instead of adding a new comment. One should not have to scroll through comments or replies to find the status of a task or sub-task.

Reviewer 2: What about something like:

```suggestion
    - Update the status of a task in the issue description instead of commenting on the issue. Use [task lists](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists) if an issue has multiple parts. This helps others who may otherwise need to scroll through comments on the issue to determine its status.
```

Feel free to improve it

    - Comments in an issue should be limited to details or context that help resolve the issue.

- If the information you provide in the issue is incomplete, then you might be contacted later during the [issue triaging process](#review_issue_to_determine_completeness_of_information).

- If you find yourself in one of the following situations, move the conversation to [MDN's discussion on GitHub](https://github.com/mdn/mdn-community/discussions):
  - A discussion needs to take place to clarify an issue.
  - A discussion begins after opening the issue.
  - The issue has no clear consensus on its resolution.
  - The requirements for completing the task expand while it's being resolved or the work is unclear.

> - You can open an issue and [fix it yourself](#fixing_issues_yourself).

Reviewer 2: I feel like this should be the first point 😄 (with hint on contributing guide)

Author: To me, the section describes how to open an issue, and then in the end says that you can open an issue and you can try to fix the issue you've opened. But good catch about linking to the contribution guide! I'll add the link to the "Fixing issues yourself" section.

> ### Creating an issue task tracker

Reviewer 2: I think we should use "task list" terminology instead of "tracker" so that we reflect GitHub terminology and docs https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists

Ensure that you list all the actionable tasks as a checklist.

For example:

   ```markdown
   // Issue title
   Ensure sections follow the order defined in the CSS property template

   // Issue description
   The CSS property page template is defined [here](/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/CSS_property_page_template).
   This issue tracker will be used to compare the documented CSS properties with the template and track changes to the property pages for compliance.

   ### List of pages checked

   - [x] [accent-color](/en-US/docs/Web/CSS/accent-color) - checked, okay
   - [ ] [backdrop-filter](/en-US/docs/Web/CSS/backdrop-filter)
   - [ ] [letter-spacing](/en-US/docs/Web/CSS/letter-spacing) - open pull request to move `Accessibility concerns` and `Internationalization concerns` sections before the `Specifications` section.
   ```

## Guidelines for working on an issue

Remember that if you take on an issue, the expectation is for the work to be completed in a timely manner. If you can no longer complete the required task, leave a comment and unassign yourself from the issue.

These are the general steps for working on an issue:

> 1. **Find an issue:** If you're looking to contribute, search for issues with `help wanted` or `good first issue` label. Most repositories have issues with these labels. You are welcome to browse and pick an issue that is suitable for your skill set.

Reviewer: ```suggestion
1. **Find an issue:** If you're looking to contribute, search for issues with the `help wanted` or `good first issue` label. Most repositories have issues with these labels. You are welcome to browse and pick an issue that is suitable for your skill set.
```

We could also link to the contributor task board https://github.com/orgs/mdn/projects/25

Author: Nice one. Added in https://github.com/mdn/temp-processes/pull/7/commits/08fa6e71762d87e1906427f51e02e7347a46a633, line 95.

   > **Note:** An issue with the `needs triage` label indicates that the MDN core team has not reviewed the issue yet, and you shouldn't begin work on it.

> 2. **Assign yourself to the issue:** After finding an issue you'd like to work on, make sure no one else is assigned to the issue. Add a comment saying you would like to work on it, and assign the issue to yourself.

Reviewer: ```suggestion
2. **Assign yourself to the issue:** After finding an issue you'd like to work on, make sure no one else is assigned to the issue. Add a comment saying you would like to work on it and assign the issue to yourself if you can do so.
```

3. **Do the research:** Most issues need some investigation before work can start.

>    - Scope out the work that needs to be done. If you need to ask questions, ask for help in the [MDN Web Docs chat room](https://chat.mozilla.org/#/room/#mdn:mozilla.org) on [Matrix](https://wiki.mozilla.org/Matrix).

Reviewer: Note to self: We should update these to refer to Discord post 17 January 2023.

   - If the issue is well-described, and the work is pretty obvious, go ahead and do it.
   - If the issue is not well-described, and/or you are not sure what is needed, feel free to @mention the poster and ask for more information.

>    - If you are not still sure who to ask, ask for help in the [MDN Web Docs chat room](https://chat.mozilla.org/#/room/#mdn:mozilla.org) on [Matrix](https://wiki.mozilla.org/Matrix).

Reviewer: I believe we are stating this twice?

Author: Yes 🙂, removed one

> 4. **Make the changes:** Fork and branch the repository. Do your work and open a [pull request](/en-US/docs/MDN/Community/Pull_requests) in the repository. Comment in the issue and ask for a review of your pull request.

Reviewer: I wonder if the last piece is needed. On most of our repositories we auto-assign reviewers or we have a codeowners file so, someone should be notified automatically.

Author: True. Fixed in https://github.com/mdn/temp-processes/pull/7/commits/b7818128a954d44ffe321092f1f136dd2012a071 line 107.

>    If you no longer have the time to work on the assigned issue, let the team know in a comment so that the issue can be assigned to someone else.

Reviewer: I guess at this point the pull request is open so, they should add the comment on the pull request, correct?

Author: Correct. Clarified in https://github.com/mdn/temp-processes/pull/7/commits/608012af784392337781acfe9b4c5a875309f291

> 5. After your pull request has been reviewed and merged, you can mark the linked issue as closed. If you opened the pull request with `Fixes #<issue>` verbiage, the issue will be closed automatically when the pull request is merged.

Reviewer: Should we link to the GitHub docs explaining this some more?

Author: Agree, I added this as part of the previous commit https://github.com/mdn/temp-processes/commit/b7818128a954d44ffe321092f1f136dd2012a071, line 107.

### Fixing issues yourself

If you spot a bug — whether it's a problem with site infrastructure or an error in documentation — you can try to fix it yourself.

> [Report the problem](#guidelines_for_reporting_an_issue) as described earlier.

Reviewer 2: Nice, we should also hint that you should assign yourself to the issue or comment / indicate that you want to fix it / start working on it.

Author: I also added that after opening the issue, you can indicate your intent to work on the issue, but wait for it to be triaged. After triaging, assign yourself to the issue to start working on it.

You can fix the problem yourself by updating the appropriate source:

- The MDN content (in English) is found in the [content](https://github.com/mdn/content) repository.
- The MDN content translated in other locales is found in the [translated-content](https://github.com/mdn/translated-content) repository.

> - The MDN platform code, which renders the content as MDN, is found in the [yari](https://github.com/mdn/yari) repository.

Reviewer: ```suggestion
- The MDN platform code, which renders the MDN Web Docs website content, is found in the [yari](https://github.com/mdn/yari) repository.
```

- In the [mdn/content repository](https://github.com/mdn/content/issues), an additional `Content:` label, such as `Content:CSS` or `Content:WebAPI`, is set on the issue automatically. This gets set based on the MDN URL mentioned in the issue. You can use the content-specific label to look for issues to be triaged in your specific topic area.

- If an issue concerns an active, non-en-US locale, set the appropriate label, such as `l10n-fr`, `l10n-zh`, or `l10n-ja`. The teams for those locales will pick these issues up and triage them.

- You don't need to actively triage issues all the time. Set aside time, say 30 minutes every week, to triage issues on a regular basis in your area of responsibility. Triaging doesn't have to be done as part of a synchronous meeting or even at the same time as everyone else, but it should be done regularly to make sure that the backlog of untriaged bugs doesn't get too high.

> - Apart from triaging incoming issues every week, review the list of old bugs to see if there are any that are stalled, need closing, or are no longer relevant.

Reviewer: We can mention that these will be auto-labelled as stale after 30 days.

Author: Done.

The [idle workflow](https://github.com/mdn/workflows/blob/main/.github/workflows/idle.yml) does apply the `idle` label after 30 days.

Does that mean we need to update the `idle` label description? Currently it is:
`Issues and pull requests with no activity for three months. `

  - Check assigned issues that are still open to see if the assignee is making progress. If there is no progress after a week of being assigned, ask them if they still have time to work on the issue. If another week passes by without any progress, unassign them and leave a comment indicating that you're making the issue available for other interested contributors.
  - If a pull request has been opened to fix the issue but has not been reviewed for a week, give the reviewer a gentle ping to ask if they can get to it.

>   - If a pull request to fix the issue is waiting on review comments to be addressed after a week, then ask the submitter if they can respond to their review. If another week goes by, either fix the review comments yourself if you have time, or close the pull request and unassign the related issue.

Reviewer: ```suggestion
  - If a pull request to fix the issue is waiting on review comments to be addressed after a week, then ask the author if they can respond to their review. If another week goes by, either fix the review comments yourself if you have time, or close the pull request and unassign the related issue.
```

To keep with the naming convention on GitHub.

> ### Decide on triagers

Author: We should remove this section "Decide on triagers" with the list of names of people responsible for triaging specific topic areas. It can get outdated quickly over time and somebody will have to take the ownership to keep it up to date. This assignment is something that should be discussed internally and people should be aware of the topic areas they are responsible for to triage issues. This assignment would also very likely map to pull request review assignments.

Reviewer 2: Good idea

- WebAPI — Ruth John
- WebExt — Caitlin/WebExt team

### Issue-specific triaging tasks

These are the guidelines to follow while triaging each issue.

#### Review the issue to determine the completeness of information

Review each issue against the following checklist to ensure that the issue contains the described information for someone to start working on the bug:

- MDN URL where the problem has been found.
- The specific heading or section on the MDN page where the problem can be found.

> - URL of an example page or repository related to the bug, if appropriate.

Reviewer: Are we repeating the need for a URL in this sentence?

Author: Fixed. TBH, I am not sure what the second one is referring to.

  - Label: `p0` (will be addressed immediately)

- Major issue: This type of issue could severely affect a page's usefulness. For example, a significant amount of out-of-date information, a complex and important code example that doesn't work, a significant amount of prose that is badly written and hard to understand, or a large number of broken links.
  - Labels: `p1` (will be addressed soon) and `p2` (will be addressed soon, but higher priority items will take precedence)

- Minor issue — This is a type of improvement issue that can make the existing content better but does not affect learning or only has a minor effect on learning. Examples include typos, bad grammar, a broken link, a small amount of out-of-date information or badly-written prose, or a code snippet that doesn't work.

>   - Labels: `p3` (no visibility when the issue will be addressed)

Reviewer: Should we mention that community help here is appreciated?

Author: Fixed in https://github.com/mdn/temp-processes/pull/7/commits/89f1dce3e897d9f1c11704030f372f65f1f2ea94, line 182.

In general, critical issues should be fixed immediately and are most likely handled by MDN staff and peers.

#### Add helpful information

> If possible, add information that can help contributors to fix the issue. The information can be in the form of steps, direction, or reading resources. You can time-box this task to 5 minutes.

Reviewer: We should call out that this is especially important for `good first issues`.

Author: Done in https://github.com/mdn/temp-processes/pull/7/commits/2a42225e4e69ce3ebf6e7b6aa361110458d97e59
